central/config/datastore/singleton.go

```diff
@@ -128,30 +125,24 @@ func initialize() {
 			DownloadableReportRetentionDays:        DefaultDownloadableReportRetentionDays,
 			DownloadableReportGlobalRetentionBytes: DefaultDownloadableReportGlobalRetentionBytes,
 		}
-		needsUpsert = true
 	}
 
 	if features.UnifiedCVEDeferral.Enabled() {
 		if privateConfig.GetVulnerabilityExceptionConfig() == nil {
 			privateConfig.VulnerabilityExceptionConfig = defaultVulnerabilityDeferralConfig
 		}
-		needsUpsert = true
 	}
 
-	if features.AdministrationEvents.Enabled() {
-		if privateConfig.GetAdministrationEventsConfig() == nil {
-			privateConfig.AdministrationEventsConfig = &storage.AdministrationEventsConfig{
-				RetentionDurationDays: DefaultAdministrationEventsRetention}
+	if privateConfig.GetAdministrationEventsConfig() == nil {
+		privateConfig.AdministrationEventsConfig = &storage.AdministrationEventsConfig{
+			RetentionDurationDays: DefaultAdministrationEventsRetention,
 		}
-		needsUpsert = true
 	}
 
-	if needsUpsert {
-		utils.Must(d.UpsertConfig(ctx, &storage.Config{
-			PublicConfig:  config.GetPublicConfig(),
-			PrivateConfig: privateConfig,
-		}))
-	}
+	utils.Must(d.UpsertConfig(ctx, &storage.Config{
```

Comment:
Why this change?

Reply:
With removing the feature flag, `needsUpsert` became always true. So I removed it.